### PR TITLE
Add responsive navigation links for HOC

### PR DESCRIPTION
**Pull request by kozzi**
This updates the header nav links so that they are available even in small screens. This should make HOC fully responsive.
### Some notes
- I didn't use Bootstrap to show/hide elements because the logic was a little more complicated (e.g. I still want to show nav links on small screens, but only when the hamburger is clicked)
- Spoke with @tanyaparker about the language drop down. It is a deliberate decision to keep it at the top even on small screens.
- Also cleaned up a bunch of css style issues

responsive-headers

### CODE REVIEW

File: pegasus/sites.v3/hourofcode.com/styles/050-theme.css [hunk 1]
```diff
@@ -18,7 +18,7 @@ body {
 
 #pageheader {
   width: 100%; 
-  background-color: #000;
+  background-color: rgba(0, 178, 192, 0.5);
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/styles/050-theme.css:22]: note that rgba isn't supported on IE8.. worth testing to see what happens in that case.

File: pegasus/sites.v3/hourofcode.com/styles/060-hoc-specific.css [hunk 2]
```diff
@@ -1,24 +1,78 @@
...
+  width: 100%;
+  max-width: 100%;
+}
+#hoc-header-main {
+  margin-top: 1em;
+  text-align: right;
+}
+#language-dropdown {
+  float: right;
+  margin-left: 2em;
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/styles/060-hoc-specific.css:18]: Is it weird to be mixing em and px in the same definitions/file?

File: pegasus/sites.v3/hourofcode.com/styles/060-hoc-specific.css [hunk 3]
```diff
@@ -103,4 +178,14 @@ label.has-error {
...
+    width: 6em;
+    position: relative;
+    top: -1em;
+    left: 0;
+    float: left;
+  }
+  #language-dropdown {
+    margin-left: 1em;
+  }
+}
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/styles/060-hoc-specific.css:205]: I wonder if it's worth pulling this much styling out into its own file so it's a bit more self-contained?

File: pegasus/sites.v3/hourofcode.com/views/theme_common_head_after.haml [hunk 4]
```diff
@@ -11,3 +11,18 @@
...
+    // When nav hamburger is clicked, toggle responsive menu
+    $("#hoc-nav-hamburger").click(function(e){
+        $('#hoc-nav ul').toggleClass('hide-responsive-menu');
+        e.preventDefault();
+    });
+
+    // When clicking anywhere outside the responsive menu, dismiss the menu
+    $(document).click(function(event) {
+      if(!$(event.target).closest('#hoc-nav-hamburger').length) {
+        $('#hoc-nav ul').addClass('hide-responsive-menu');
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/views/theme_common_head_after.haml:25]: Interesting technique.. have we used it elsewhere?  does it work okay for touch devices?

**Comment by breville** [comment]
Looks nice in action, and no major issues from looking through the code.  It'll be important to test this on a ton of devices, resolutions, and locales.

We might just have to ponder bringing this over to code.org as well!

File: pegasus/sites.v3/hourofcode.com/styles/050-theme.css (hunk 1 above)
Comment (kozzi) [inline pegasus/sites.v3/hourofcode.com/styles/050-theme.css:22]: This was originally inline CSS on the div; I simply moved it to the stylesheet. Presumably that shouldn't change the behaviour, but will definitely do more testing!

File: pegasus/sites.v3/hourofcode.com/styles/060-hoc-specific.css (hunk 2 above)
Comment (kozzi) [inline pegasus/sites.v3/hourofcode.com/styles/060-hoc-specific.css:18]: The height of the teal horizontal header bar was defined in em originally (before this change), and I wanted to position the branding logo vertically centered with this bar, which is why I used em here, to be consistent with the units of the header bar itself.

File: pegasus/sites.v3/hourofcode.com/views/theme_common_head_after.haml (hunk 4 above)
Comment (kozzi) [inline pegasus/sites.v3/hourofcode.com/views/theme_common_head_after.haml:25]: Hm, not that I'm aware of yet. Will do a code search, and will test on touch devices!

Edit: Verified to work on iPhone!

File: pegasus/sites.v3/hourofcode.com/styles/060-hoc-specific.css (hunk 3 above)
Comment (kozzi) [inline pegasus/sites.v3/hourofcode.com/styles/060-hoc-specific.css:205]: Sure! Will probably need a bit more guidance on when it makes sense to pull out styling into its own file. I'll find you to chat more about this offline when I come in! Cause I noticed there's also a bunch of stylesheets in a separate location (/public/css), and would love background on why some belonged there.

File: pegasus/sites.v3/hourofcode.com/styles/050-theme.css (hunk 1 above)
Comment (kozzi) [inline pegasus/sites.v3/hourofcode.com/styles/050-theme.css:22]: Okay, just tested this. It indeed does not work in IE8. Looks like the stylesheet sets the "fallback" color (#000) and then the transparent rgba color is set inline in the html. Fixing this with my next check-in.

**Comment by kozzi** [comment]
Yeah, definitely something to think about bringing it over for code.org. With code.org though, I have additional work to first make sure all the individual pages are responsive first (they aren't currently). I'll look into that after this.

**Comment by kozzi** [comment]
Did a round of testing on IE8/IE9:

### Before

(The logo currently doesn't show up in production for IE8. We must have broken that a while back.)
image
image

### After

image
image

Notable differences (which I think are fine fallbacks):
- Header bar is black
- There's no "arrow" pointing to the hamburger menu when the nav menu is open

**Comment by breville** [comment]
These seem reasonable for such old browsers.  It's possible to have a semi-transparent bar on those browsers but it requires a combination of opacity & an non-rgba background-color.

**Comment by breville** [comment]
I think you can do both an rgb and rgba entry to have the bar show solid teal on ie8/9 and transparent teal on newer browsers.  I think that'd be a nice simple improvement.  Everything else lgtm from prior reviews.

**Pull request merged**